sdk/ai/azure-ai-projects/azure/ai/projects/models/_patch.py

```diff
@@ -0,0 +1,1369 @@
+    MessageTextFileCitationAnnotation,
+    MessageTextFilePathAnnotation,
+)
+
+from abc import ABC, abstractmethod
+from typing import (
+    AsyncIterator,
+    Awaitable,
+    Callable,
+    List,
+    Dict,
+    Any,
+    Type,
+    Optional,
+    Iterator,
+    Tuple,
+    Set,
+    get_origin,
+    cast,
+    get_args,
+    Union,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def _filter_parameters(model_class: Type, parameters: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Remove the parameters, non present in class public fields; return shallow copy of a dictionary.
+
+    **Note:** Classes inherited from the model check that the parameters are present
+    in the list of attributes and if they are not, the error is being raised. This check may not
+    be relevant for classes, not inherited from azure.ai.projects._model_base.Model.
+    :param model_class: The class of model to be used.
+    :param parameters: The parsed dictionary with parameters.
+    :return: The dictionary with all invalid parameters removed.
+    """
+    new_params = {}
+    valid_parameters = set(
+        filter(
+            lambda x: not x.startswith("_") and hasattr(model_class.__dict__[x], "_type"), model_class.__dict__.keys()
+        )
+    )
+    for k in filter(lambda x: x in valid_parameters, parameters.keys()):
+        new_params[k] = parameters[k]
+    return new_params
+
+
+def _safe_instantiate(model_class: Type, parameters: Union[str, Dict[str, Any]]) -> Any:
+    """
+    Instantiate class with the set of parameters from the server.
+
+    :param model_class: The class of model to be used.
+    :param parameters: The parsed dictionary with parameters.
+    :return: The class of model_class type if parameters is a dictionary, or the parameters themselves otherwise.
+    """
+    if not isinstance(parameters, dict):
+        return parameters
+    return model_class(**_filter_parameters(model_class, parameters))
+
+
+class ConnectionProperties:
+    """The properties of a single connection.
+
+    :ivar id: A unique identifier for the connection.
+    :vartype id: str
+    :ivar name: The friendly name of the connection.
+    :vartype name: str
+    :ivar authentication_type: The authentication type used by the connection.
+    :vartype authentication_type: ~azure.ai.projects.models._models.AuthenticationType
+    :ivar connection_type: The connection type .
+    :vartype connection_type: ~azure.ai.projects.models._models.ConnectionType
+    :ivar endpoint_url: The endpoint URL associated with this connection
+    :vartype endpoint_url: str
+    :ivar key: The api-key to be used when accessing the connection.
+    :vartype key: str
+    :ivar token_credential: The TokenCredential to be used when accessing the connection.
+    :vartype token_credential: ~azure.core.credentials.TokenCredential
+    """
+
+    def __init__(
+        self, *, connection: GetConnectionResponse, token_credential: Optional[TokenCredential] = None
```

Comment:
Does this accept both TokenCredential and AsyncTokenCredential?

Reply:
@dargilco   I checked the code.  It seems that we pass AsyncTokenCredential.    I am not sure if we should use Union[TokenCredential, AsyncTokenCredential, None] here.

Reply:
Good call @kristapratico! I changed the type to `token_credential: Union[TokenCredential, AsyncTokenCredential, None]` . FYI @howieleung